src/Raven.Server/Documents/Archive/ArchivalStorage.cs

```diff
@@ -0,0 +1,356 @@
+        {
+            if ((doc.Flags & DocumentFlags.HasCounters) == 0) return;
+            
+            var counterGroups = _database.DocumentsStorage.CountersStorage.GetCounterValuesForDocument(context, documentId).ToList();
+                                            
+            var clonedCountersGroupsValues = new List<BlittableJsonReaderObject>(counterGroups.Count);
+            foreach (var counterGroup in counterGroups)
+                clonedCountersGroupsValues.Add(counterGroup.Values.Clone(context));
+                                            
+            var mergedCounterGroupsValues = BuildMergedCounterGroupValues(context, clonedCountersGroupsValues, documentId);
+                                                
+            _database.DocumentsStorage.CountersStorage.DeleteCountersForDocument(context, documentId, new CollectionName(collectionName));
+                                            
+            var newCounterEtag = _database.DocumentsStorage.GenerateNextEtag();
+            var newCounterChangeVector = _database.DocumentsStorage.GetNewChangeVector(context, newCounterEtag);
+            _database.DocumentsStorage.CountersStorage.PutCounters(context, documentId, archivedCollectionName, newCounterChangeVector , mergedCounterGroupsValues, newCounterEtag);
+        }
+
+        private void ArchiveDocumentTimeSeriesIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+            
+        }
+        
+        private void ArchiveDocumentRevisionsIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+            
+        }
+        
+        private void ArchiveDocumentAttachmentsIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+                    
+        }
+
+        private bool MigrateDocument(DocumentsOperationContext context, (Slice LowerId, string Id) ids, DateTime currentTime)
+        {
+            if (ids.Id == null)
+                throw new InvalidOperationException($"Couldn't archive the document. Document id is null. Lower id is {ids.LowerId}");
+
+            using (var doc = _database.DocumentsStorage.Get(context, ids.LowerId, DocumentFields.Data, throwOnConflict: true))
+            {
+                if (doc == null || !doc.TryGetMetadata(out var metadata))
+                {
+                    throw new InvalidOperationException($"Failed to fetch the metadata of document '{ids.Id}'");
+                }
+                if (HasPassed(metadata, currentTime) == false) return false;
+                        
+                var collectionName = metadata[Constants.Documents.Metadata.Collection].ToString();
+                string archivedCollectionName = $"{collectionName}Archived";
```

Comment:
Do we want to give the user some choice here?
What if they are using non English collection names? Or something that isn't word like?

Reply:
I've decided to add an "Archived" postfix because I was aware of issues about sorting collections by name. I see no evil in giving users control of this suffix. It may be 'Archived' by default, but customizable.

Reply:
+1 for making it customizable